src/api-client.ts

```diff
@@ -1,14 +1,25 @@
 import { IAuthProvider } from "./iauth-provider";
 import { RequestOptions } from "./types";
 import axios, { AxiosInstance } from "axios";
+import FireBlocksError from "./fireblocks-error";
 
 export class ApiClient {
     private axiosInstance: AxiosInstance;
 
-    constructor(private authProvider: IAuthProvider, private apiBaseUrl: string, private options: {timeoutInMs?: number}) {
+    constructor(private authProvider: IAuthProvider, private apiBaseUrl: string, private options: { timeoutInMs?: number }) {
         this.axiosInstance = axios.create({
             baseURL: this.apiBaseUrl
         });
+        this.axiosInstance.interceptors.response.use(
+            (response) => response,
+            (error) => {
+                if (error?.response?.data) {
+                    throw new FireBlocksError(error.response.data.message, error.response.data.code, error.response.status);
```

Comment:
this is a breaking change since clients expect axios errors isn't it?

Reply:
@yarinvak hi 
Thanks for the review
No we don't have access to the data that fireblocks backend returns as error in the axios error 
that's why I added this FireBlocksError and it has all the data that fireblocks returns in case of error

and No it's not a breaking change as no one is getting any info on error messages right now, only the status code
since FireBlocksError extends Error class they will get what they used to get but with some additional info that can help them debug and communicate with support better

Reply:
it's worth mentioning we are using this code for our own application and it was a solution for not getting enough info on 400 errors :)